fontforge/splineoverlap.c

```diff
@@ -843,8 +845,13 @@ return;
         else if (Within16RoundingErrors(m->tstart, m->tend))
 	    SOError( "Attempt to subset monotonic rejoin inappropriately: m->tstart and m->tend are very close (%f = %f, t = %f)\n",
 		    m->tstart, m->tend, t );
-	else if (Within4RoundingErrors(m->s->from->me.x,m->s->to->me.x) && Within4RoundingErrors(m->s->from->me.y,m->s->to->me.y))
+	else if (m->s->from->nonextcp && m->s->to->noprevcp &&
+		Within4RoundingErrors(m->s->from->me.x,m->s->to->me.x) &&
+		Within4RoundingErrors(m->s->from->me.y,m->s->to->me.y))
 	    SOError( "The curve is too short.\n");
```

Comment:
About a month ago I read enough about actual spline math that I have started to understand this stuff a little, including the macro above (relative to FontForge's representation, which I understand bits of). I take it here you're replacing a dodgy heuristic of just looking at the end point locations with an evaluation of the endpoints in the line case (because nothing interesting can happen between the very-close-together endpoints of a line) and the spline case (where far-away control points might stretch the spline enough away from the endpoints to make the t-interval "subcurve" interesting).

If that's right, might "The line is too short." be a more informative message in the former case?

Reply:
Agreed. Or maybe "This is a line segment, and it is too short.".